### ISSUE TITLE
Fix `declaration-block-no-shorthand-property-overrides` false negatives for logical properties and physical shorthands

### ISSUE THREAD

**Issue by woody-li** [stylelint/stylelint#9418]
### What minimal example or steps are needed to reproduce the bug?

```css
padding-inline: 5px;
padding: 10px;
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-shorthand-property-overrides": true
  }
}
```

### How did you run Stylelint?

All ways.

### Which Stylelint-related dependencies are you using?

"stylelint": "17.14.1"

### What did you expect to happen?

Report problem:
```
Overridden property "padding-inline" by shorthand "padding" (declaration-block-no-shorthand-property-overrides)
```

### What actually happened?

No violation is reported.

**Comment by jeddy3** [comment]
@woody-li Thanks for the report and for using the template.

It'd be good to get some specifications and browser behaviour references for how logical properties and shorthand physical ones interact. Can you help us with this?

You may want to disallow shorthand physical properties using the [`property-disallowed-list`](https://stylelint.io/user-guide/rules/property-disallowed-list) rule if you are favouring logical properties and values.
